C7/UIElements/Popups/Popup.cs

```diff
@@ -105,6 +101,14 @@ protected void AddBackground(int width, int height, int vOffset = 0)
 
     private TextureRect CreateBackground(int width, int height)
     {
+        TextureRect rect = new TextureRect();
+
+        if (backgroundCache.ContainsKey((width, height)))
```

Comment:
Just wondering, were you able to notice a difference with this caching?  I always like to be able to document if a performance improvement made a difference, since otherwise it's easy to chase things that don't yield improvements but might increase complexity, and occasionally I've even made what I thought would be an improvement, but actually made things worse.

I don't see any problem with having this cache, so it doesn't matter much, just something to think about.

Reply:
On my laptop the initial creation of popups spends ~2ms on creating the background texture, which is bypassed on subsequent renderings of popups with the same dimensions. I can't really notice a difference running it, but it does cut out those 2ms. I added the optimization because there's a comment in the code somewhere saying that it's expensive to create the background with the current method. If the 2ms isn't worth the added complexity I can remove.

Reply:
Cool, good to hear there are some savings.  I wonder if that comment is from the very first iteration of popups?  It used to be a lot worse.  Either way, that's 1/8th of a 60 FPS target saved, so it adds up.